tests/test_switch.py

```diff
@@ -928,3 +931,39 @@ async def test_area(hass):
         switch.turn_on_off_listener.last_service_data,
     )
     assert light.entity_id not in switch.turn_on_off_listener.last_service_data
+
+
+async def test_change_switch_settings_service(hass):
```

Comment:
@th3w1zard1 I added this test :)

Reply:
Nice work! `change_switch_settings` probably doesn't need any error checking either as everything is validated through Home Assistant. I was worried about users trying to set the interval to 0 or something, but since that option was removed I doubt anything more needs to be done here.